### ISSUE TITLE
How is the reshaping possible?

### ISSUE THREAD

**Issue by aneesh-joshi** [shmsw25/qa-transfer#2]
Hi @shmsw25,
Thank you for your work. I have been trying to reproduce it for a while.

Please could you help me understand:
In the line below, you define 

https://github.com/shmsw25/qa-transfer/blob/b5d62c52d5b6f290091128ee64338fe6f1498a08/basic/model.py#L226

But earlier, while maxpooling, you define:
https://github.com/shmsw25/qa-transfer/blob/b5d62c52d5b6f290091128ee64338fe6f1498a08/basic/model.py#L213-L215

If `g2` is of size `[N, M, 2d]`

How are you able to later resize it to 
`N, g_dim` = `N, 2*d`

Your comments will be appreciated.
Thanks you!

**Comment by shmsw25** [comment]
Hi.
I'm sorry about the confusing part in the code. Actually, M=1 in the entire code. Originally it meant to be the number of paragraph, but in our problem setting, the number of paragraph is always 1. Therefore the reshape is possible.
Thanks!

**Comment by aneesh-joshi** [comment]
Thank you for that clarification
Could you also share in which split the results of WikiQA MAP score are given?
Is it the Dev, train, test or full dataset?
@shmsw25

**Comment by shmsw25** [comment]
It is the result on the test set!

**Comment by aneesh-joshi** [comment]
@shmsw25 
That's awesome.
I have been trying to reproduce the results for a few weeks now. (only getting 0.64 MAP on WikiQA test set even after pretraining on SQUAD-T with BiDAF-T)
Here's my implementation : https://github.com/aneesh-joshi/BiDAF/blob/master/nh.py
I'm sure you're very busy. It would be great if you could take a look.
